### PR TITLE
core: get infra instead of load the infra in simulation and results

### LINKED ISSUE #3185: Add a train to the path different from the path of the first train, the train does not appear in the list
### What happened?

When a train is added which has a different path from the first added train, it does not appear in the list of trains in the scenario. 

Here are the line in the code that causes this issue : https://github.com/DGEXSolutions/osrd/blob/3dbd56326a0bf67d9b2902f1017295ec17c19250/api/osrd_infra/views/train_schedule/train_schedule.py#L111-L112

### What did you expect to happen?

When adding a train with a different path, it should be displayed in the train list but its GET should not be added.

### How can we reproduce it (as minimally and precisely as possible)?

1. Create a Scenario
2. Add a train 
3. Add a second train with a different path
4. You will see that your second train won't be added

### What operating system and browser are you using? (if applicable)

- Macbook M1
- Google Chrome 22

### OSRD version

1.0

### LINKED ISSUE #4341: front + back: return an error in the runing time calculation and simulation if the infrastructure is not loaded
# AC

## Scenario

- [ ] train list should now use the new timetable endpoint providing train schedule details and therefore be independant of simulation results
- [ ] train list should be displayed even when the infra is loading
- [ ] train list should be dynamically updated when adding or deleting a train
- [ ] train list should be dynamically updated when modifying a train (in the managetrainschedule or in the simulation results when moving a train)  

## Simulation result

- [ ] front : 2 calls back to list trains + display simulation result
- [ ] back : return an error in the simulation result if infra not loaded

## Simulation : 

- [ ] back :  return an error if infra not loaded 
- [ ] front : grey the button add a train when the infra is not loaded

## STDCM

- [ ] front : grey the button "add" in STDCM when infra is not loaded

## Implem Idea

- [ ] fusionner le CSS pour le loader dans common

### LINKED ISSUE #4447: front: sort trains by chronological order in scenarios's train list
### Description and goal

To improve comprehension and readability for users, trains should be sorted by chronological order in a scenario train list.

- Beware as this change requires to change the logic of all `selectedTrain` state in the app
- Maybe it would be interesting to created an enhanced editoast endpoint for timetable which automaticly sort the trains like it has been done for the rollingstocks

### Acceptance criteria

- [ ] Trains in a scenario train list should be sorted by chronological order
- [ ] Selection of a train in the list should still select the good train on the GET
- [ ] Add/remove a train should still reflect on the GET

### PR DESCRIPTION
**Pull request by younesschrifi**
Close #4341 
close #3185 
close #4447

### CODE REVIEW

File: editoast/src/models/timetable.rs [hunk 1]
```diff
@@ -64,6 +74,66 @@ impl Timetable {
...
+                        .head_positions
+                        .last()
+                        .expect("Train should have at least one position")
+                        .time
+                        + train_schedule.departure_time;
+                    let eco = if let Some(eco) = result_train_eco {
+                        Some(eco.0.mechanical_energy_consumed)
+                    } else {
+                        None
+                    };
```
Comment (leovalais) [inline editoast/src/models/timetable.rs:90]:
```suggestion
                    let eco = result_train_eco.map(|eco| eco.0.mechanical_energy_consumed);
```

File: editoast/src/models/timetable.rs [hunk 2]
```diff
@@ -64,6 +74,66 @@ impl Timetable {
...
+                    let mechanical_energy_consumed = MechanicalEnergyConsumedBaseEco {
+                        base: result_train.mechanical_energy_consumed,
+                        eco,
+                    };
+                    let filtered_stops: Vec<ResultStops> = result_train
+                        .stops
+                        .into_iter()
+                        .filter(|stop| stop.duration > 0 as f64)
```
Comment (leovalais) [inline editoast/src/models/timetable.rs]:
```suggestion
                        .filter(|stop| stop.duration > 0.)
```

File: editoast/src/views/mod.rs [hunk 3]
```diff
@@ -35,8 +35,8 @@ pub fn routes() -> impl HttpServiceFactory {
         timetable::routes(),
         rolling_stocks::routes(),
         light_rolling_stocks::routes(),
-        pathfinding::routes(),
         train_schedule::routes(),
+        pathfinding::routes(),
```
Comment (leovalais) [inline editoast/src/views/mod.rs]: You probably can revert this change 😁

File: editoast/src/models/timetable.rs [hunk 4]
```diff
@@ -64,6 +74,66 @@ impl Timetable {
         .unwrap()
     }
 
+    /// Retrieves timetable with a specific id, its associated train schedules details and
+    /// some informations about the simulation result
+    pub async fn with_train_schedules_with_details(
```
Comment (leovalais) [inline editoast/src/models/timetable.rs]: If `Timetable::with_train_schedules` is not used anymore better delete it.

File: editoast/src/views/train_schedule/mod.rs [hunk 5]
```diff
@@ -66,17 +64,13 @@ pub enum TrainScheduleError {
...
-        get_results,
-    ))
+    web::scope("/train_schedule")
+        .service(get_results)
+        .service(scope("/{id}").service((get, patch, standalone_simulation, get_result)))
 }
 
 /// Return a specific timetable with its associated schedules
-#[get("")]
+#[get("/{id}")]
```
Comment (leovalais) [inline editoast/src/views/train_schedule/mod.rs]: I think it's not necessary to add the `{id}` here since it's induced by the `scope()` above.

File: editoast/src/views/train_schedule/simulation_report.rs [hunk 6]
```diff
@@ -95,7 +95,6 @@ pub async fn create_simulation_report(
     } else {
         None
     };
-
```
Comment (leovalais) [inline editoast/src/views/train_schedule/simulation_report.rs]: Either remove this change or also delete the other line breaks in the function.

File: editoast/src/models/train_schedule.rs [hunk 7]
```diff
@@ -115,6 +115,22 @@ pub struct TrainScheduleDetails {
...
 }
 
+#[derive(Serialize, Debug, PartialEq, Clone, Default)]
+pub struct MechanicalEnergyConsumedBaseEco {
+    pub base: f64,
+    pub eco: Option<f64>,
+}
+
+#[derive(Serialize, Debug, PartialEq, Clone, Queryable)]
+pub struct TrainScheduleWithDetails {
```
Comment (leovalais) [inline editoast/src/models/train_schedule.rs]: The structs `TrainScheduleDetails` and `TrainScheduleWithDetails` almost have the same name but different contents/usages. Maybe rename one of them and add a small doc comment to explain how/where they're supposed to be used and what they mean.

**Review by leovalais — CHANGES_REQUESTED** [review]
I have proof read the back only, is that PR still a draft of can I test it now?

File: editoast/openapi.yaml [hunk 8]
```diff
@@ -1441,19 +1441,10 @@ paths:
                     type: number
                   name:
                     type: string
-                  train_schedules:
+                  train_schedules_with_details:
```
Comment (multun) [inline editoast/openapi.yaml]:
```suggestion
                  detailed_train_schedules:
```

**Comment by leovalais** [comment]
Just tested GET /timetable/X

**Review by leovalais — APPROVED** [review]
LGTM (editoast)

File: core/src/main/java/fr/sncf/osrd/api/InfraManager.java [hunk 9]
```diff
@@ -160,6 +160,7 @@ private FullInfra downloadInfra(
                     throw new UnexpectedHttpResponse(response);
 
                 // Parse the response
+
```
Comment (Khoyo) [inline core/src/main/java/fr/sncf/osrd/api/InfraManager.java]: Spurious newline

File: core/src/main/java/fr/sncf/osrd/api/SignalProjectionEndpoint.java [hunk 10]
```diff
@@ -52,10 +52,9 @@ public Response act(Request req) throws Exception {
             var request = adapterRequest.fromJson(body);
             if (request == null)
                 return new RsWithStatus(new RsText("missing request body"), 400);
-
-            // load infra
-            // TODO : change with get infra when the front is ready
-            var infra = infraManager.load(request.infra, request.expectedVersion, recorder);
+            System.out.println(request.expectedVersion);
```
Comment (flomonster) [inline core/src/main/java/fr/sncf/osrd/api/SignalProjectionEndpoint.java:55]: We should use logger here

File: tests/tests/test_timetable.py [hunk 11]
```diff
@@ -81,14 +93,15 @@ def test_get_timetable(
     timetable_id = timetable_id or small_scenario.timetable
 
     response = requests.get(f"{service_url}timetable/{timetable_id}/")
+    print(response.json())
```
Comment (flomonster) [inline tests/tests/test_timetable.py]:
```suggestion
```

**Review by flomonster — APPROVED** [review]
LGTM

File: front/public/locales/fr/simulation.json [hunk 12]
```diff
@@ -2,8 +2,10 @@
   "allowances": "Marges",
   "departureTime": "Départ",
   "errorMessages": {
-    "unableToRetrieveTrainSchedule": "Calcul de marche impossible"
+    "unableToRetrieveTrainSchedule": "Calcul de marche impossible",
+    "unableToLoadInfra": "Erreur de chargement de l'infra"
```
Comment (nicolaswurtz) [inline front/public/locales/fr/simulation.json]: If "unable to" it should be translated to « impossible de charger l'infrastructure »
Unless it's an error, so key should be `errorLoadingInfra`
THE CHOICE IS YOURS :D

File: tests/tests/test_timetable.py (hunk 11 above)
Comment (eckter) [reply tests/tests/test_timetable.py]: Is there a reason why it's resolved? I would have commented the same thing

File: tests/tests/test_timetable.py [hunk 13]
```diff
@@ -14,14 +14,27 @@ class _TrainScheduleDetails:
...
+    timetable_id: int
+    rolling_stock_id: int
+    initial_speed: int
+    labels: Iterable[str]
+    allowances: Iterable[dict]
+    speed_limit_tags: str
+    comfort: str
+    options: dict
+    power_restriction_ranges: Iterable[dict]
+    mechanical_energy_consumed: Iterable[dict]
```
Comment (eckter) [inline tests/tests/test_timetable.py]: The `Iterables` don't generally support `len(x)`, which we use later on. It should probably be changed to `List`

File: tests/tests/test_timetable.py [hunk 14]
```diff
@@ -98,16 +111,51 @@ def test_get_timetable(
...
+            ["new train", "west", "south east"],
+            [],
+            None,
+            "STANDARD",
+            None,
+            None,
+            {"base": 6041639237.0, "eco": None},
+            0,
+            45549.5653000392,
+        ),
```
Comment (eckter) [inline tests/tests/test_timetable.py]: Some types don't match, it causes some warnings in pycharm.

The int/float and None are not that important, but the "expected `Iterable[Dict]`, got `Dict[...]` instead" is more worrying

File: tests/tests/test_timetable.py [hunk 15]
```diff
@@ -14,14 +14,27 @@ class _TrainScheduleDetails:
...
+    stops: int
+    path_length: float
 
 
 @dataclass(frozen=True)
 class _TimetableResponse:
     id: int
     name: str
-    train_schedules: Iterable[_TrainScheduleDetails]
+    detailed_train_schedules: Iterable[_TrainScheduleDetails]
```
Comment (eckter) [inline tests/tests/test_timetable.py]: At this point it's still an iterable of dicts, it's converted to `_TrainScheduleDetails` later on

**Review by eckter — COMMENTED** [review]
I only reviewed core and tests, and only tested the tests

File: tests/tests/test_timetable.py (hunk 11 above)
Comment (leovalais) [reply tests/tests/test_timetable.py]: I fixed it in https://github.com/osrd-project/osrd/commit/279df9656859a3fb90e0e07cf12860a0d5a9cf26 but somehow the commit doesn't show here 🤔

Comment (leovalais) [reply tests/tests/test_timetable.py]: Well, now the commits has decided to show himself so I'm resolving 😁

File: tests/tests/test_timetable.py (hunk 14 above)
Comment (leovalais) [reply tests/tests/test_timetable.py]: Those should be fixed now.

**Review by eckter — APPROVED** [review]
LGTM on the core and test changes

File: editoast/src/models/train_schedule.rs [hunk 16]
```diff
@@ -107,14 +107,32 @@ impl From<TrainScheduleChangeset> for TrainSchedule {
...
 
+#[derive(Serialize, Debug, PartialEq, Clone, Default)]
+pub struct MechanicalEnergyConsumedBaseEco {
+    pub base: f64,
+    pub eco: Option<f64>,
+}
+
+/// Returns the timetable with the train's important route information (distance travelled, arrival time, etc.)
+#[derive(Serialize, Debug, PartialEq, Clone, Queryable)]
+pub struct DetailedTrainSchedule {
```
Comment (Khoyo) [inline editoast/src/models/train_schedule.rs]: TrainScheduleDetails wasn't a good name, but I think this is worse. It's not detailed, it's a summary (a real TrainSchedule has *more details*)

The timetable is detailed, not the train schedule.

Comment (leovalais) [reply editoast/src/models/train_schedule.rs]: Is `TrainScheduleSummary` better?

Comment (Khoyo) [reply editoast/src/models/train_schedule.rs]: You got my drift ;)

File: front/public/locales/fr/simulation.json (hunk 12 above)
Comment (nicolaswurtz) [reply front/public/locales/fr/simulation.json]: Infrastructure en toutes lettres ;)

File: front/src/applications/operationalStudies/components/Scenario/Timetable.tsx [hunk 17]
```diff
@@ -148,25 +159,68 @@ export default function Timetable({
...
+
+  useEffect(() => {
+    if (timetableID && reloadTimetable) {
+      getTimetableWithTrainSchedulesDetails({ id: timetableID })
+        .unwrap()
+        .then((result) => {
+          setTimetable(result);
+        });
+    }
+  }, [timetableID, reloadTimetable]);
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/Scenario/Timetable.tsx]: You could simplify and avoid 3 useEffect I think, try to fire `setTrainsDurationsIntervals` when updating timetable after fetch, use conditions to test if `reloadTimetable` is needed, etc.

File: front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.jsx [hunk 18]
```diff
@@ -36,8 +38,12 @@ export async function changeTrain(details, id) {
         timetable: details.timetable || trainDetail.timetable,
         train_name: details.train_name || trainDetail.train_name,
       };
+      dispatch(updateReloadTimetable(true));
       await patch(`${trainscheduleURI}${id}/`, params);
+      dispatch(updateReloadTimetable(false));
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.jsx:43]: why put this here ? you should do this IN timetable updating functions

File: front/src/applications/operationalStudies/views/Scenario.tsx [hunk 19]
```diff
@@ -97,25 +100,45 @@ export default function Scenario() {
...
+        });
+    }
+  }, [reloadTimetable]);
+
+  useEffect(() => {
+    if (infraId) {
+      reloadInfra({ id: infraId as number }).unwrap();
+    }
+  }, [infraId]);
+
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/views/Scenario.tsx]: Same here, it's not possible to make it more concise ? less useEffect ! :D

File: front/src/applications/operationalStudies/views/SimulationResults.tsx [hunk 20]
```diff
@@ -102,13 +109,21 @@ export default function SimulationResults({
...
+    }
+    if ((!simulation || simulation.trains.length === 0) && !isUpdating && infraState === 'CACHED') {
+      return <h1 className="text-center">{t('simulation:noData')}</h1>;
+    }
+    return null;
+  };
 
   if (!displaySimulation || isUpdating) {
-    return <div className="pt-5 mt-5">{waitingLoader}</div>;
+    return <div className="pt-5 mt-5">{waitingMessage()}</div>;
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/views/SimulationResults.tsx]: When loading, put a spinner like « mise à jour »

File: front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx [hunk 21]
```diff
@@ -36,18 +37,37 @@ export default function OSRDConfig({
...
   const shouldDisplayStdcmResult = simulation.trains[selectedTrain] !== undefined;
 
+  useEffect(() => {
+    if (infra && infra.state === 'NOT_LOADED') {
+      setIsInfraLoaded(false);
+    }
+
+    if (infra && infra.state === 'CACHED') {
+      setIsInfraLoaded(true);
+    }
```
Comment (nicolaswurtz) [inline front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx:67]: you should use an enum to list `infra.state` states common for all test about this

File: front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx [hunk 22]
```diff
@@ -73,6 +93,7 @@ export default function OSRDConfig({
                 <button
                   className="btn btn-sm  btn-primary "
                   type="button"
+                  disabled={infra?.state !== 'CACHED'}
```
Comment (nicolaswurtz) [inline front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx:95]: same here

**Review by nicolaswurtz — CHANGES_REQUESTED** [review]

File: front/src/applications/operationalStudies/views/Scenario.tsx (hunk 19 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/Scenario.tsx:134]: Yeah, I'm not a fan either for these 3 but : 
- If i put the 3rd one in the 1st one. It means each time infra will change (and during infra load, polling refetch infra every second so it changes almost every second), it will relaunch the reloadInfra so we will have 2 calls (the reload and the polling) instead of one. The 3rd is actually just here when we enter the page to check if infra is still up to date or if it need to be reloaded.
- If i put the 2nd one in the 1st one, it means each time reloadTimetalbe will change (it always goes from false > true > false when adding/deleting/modifying a train), it will launch multiple api calls for both endpoint in the first useEffect.

If you have a better suggestion/workflow for that, I'd be very happy to try it :)

File: front/public/locales/fr/simulation.json (hunk 12 above)
Comment (SharglutDev) [reply front/public/locales/fr/simulation.json]: done :)

File: front/src/applications/operationalStudies/views/SimulationResults.tsx (hunk 20 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/SimulationResults.tsx]: done :) Like you suggested, I moved the ScenarioLoader in SimulationResults as well as the text files from scenario.json to simulation.json

File: front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.jsx (hunk 18 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.jsx:43]: Hopefully this call can be removed later when we will be using rtk query tags.

File: front/src/applications/operationalStudies/views/Scenario.tsx (hunk 19 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/Scenario.tsx:134]: Probably we will be able to remove the 2nd one when using rtk query tags

File: front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.jsx (hunk 18 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.jsx:43]: done :) Add the dispatch in duplicateTrain, deleteTrain, SubmitConfAdd and SubmitConfUpdate (transformed it into a component like SubmitAdd to use rtk in it but not with editoast as trainschedules endpoints are broken in it)

Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/simulationResultsHelpers.jsx:43]: Not sure why I had to do it with SubmitConfUpdate and not SubmitConfAdd (which works the same), but I had to put the dispatch in a Promise.resolve()

File: front/src/applications/operationalStudies/components/Scenario/Timetable.tsx (hunk 17 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/Scenario/Timetable.tsx]: done :) I move some of the logic from Scenario to Timetable. Cleaner and seems to work properly

File: front/src/applications/operationalStudies/views/Scenario.tsx (hunk 19 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/views/Scenario.tsx]: done :) Moved some of the logic in Timetable

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx [hunk 23]
```diff
@@ -69,12 +71,17 @@ import { MapLayerMouseEvent } from '../../../../types';
 function getPosition(
   positionValues: PositionValues,
   allowancesSettings?: AllowancesSettings,
-  id?: number,
+  // id?: number,
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]:
```suggestion
```

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx [hunk 24]
```diff
@@ -69,12 +71,17 @@ import { MapLayerMouseEvent } from '../../../../types';
...
+  trainId?: number,
   baseKey?: string
 ) {
-  const key = (
-    allowancesSettings && id && allowancesSettings[id]?.ecoBlocks ? `eco_${baseKey}` : baseKey
-  ) as keyof PositionValues;
+  const key =
+    // allowancesSettings && id && allowancesSettings[id]?.ecoBlocks ? `eco_${baseKey}` : baseKey
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]:
```suggestion
```

File: front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/withOSRDData.jsx [hunk 25]
```diff
@@ -103,8 +96,8 @@ const withOSRDData = (Component) =>
       dispatch(updateChart(chart));
     };
 
-    const dispatchUpdateSelectedTrain = (_selectedTrain) => {
-      dispatch(updateSelectedTrain(_selectedTrain));
+    const dispatchUpdateSelectedTrainId = (_selectedTrain) => {
+      dispatch(updateSelectedTrainId(_selectedTrain));
```
Comment (clarani) [inline front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/withOSRDData.jsx]:
```suggestion
    const dispatchUpdateSelectedTrainId = (_selectedTrainId) => {
      dispatch(updateSelectedTrainId(_selectedTrainId));
```

File: front/src/applications/stdcm/components/OldAllowances/EmptyLine.tsx [hunk 26]
```diff
@@ -32,40 +31,37 @@ function getAllowanceValue(values: RangeAllowance | EngineeringAllowance) {
...
 ) {
-  if (allowanceType === 'engineering') {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
+  const selectedTrain = useSelector(getSelectedTrain);
+
+  if (selectedTrain) {
+    if (allowanceType === 'engineering') {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```
Comment (clarani) [inline front/src/applications/stdcm/components/OldAllowances/EmptyLine.tsx:42]:
```suggestion
```

File: front/src/applications/stdcm/views/StdcmRequestModal.jsx [hunk 27]
```diff
@@ -79,18 +79,16 @@ export default function StdcmRequestModal(props) {
...
               KEY_VALUES_FOR_CONSOLIDATED_SIMULATION,
-              newSimulation.trains,
+              trains,
               t
             );
             dispatch(updateConsolidatedSimulation(consolidatedSimulation));
-            dispatch(updateSimulation(newSimulation));
-            dispatch(updateSelectedTrain(newSimulation.trains.length - 1));
+            dispatch(updateSimulation({ trains }));
+            dispatch(updateSelectedTrainId(trains[0].id));
```
Comment (clarani) [inline front/src/applications/stdcm/views/StdcmRequestModal.jsx]: This is for STDCM projection :)
```suggestion
            dispatch(updateSelectedTrainId(fakeNewTrain.id));
```

File: front/src/common/api/osrdEditoastApi.ts [hunk 28]
```diff
@@ -1678,6 +1670,87 @@ export type ProjectPatchRequest = {
...
+  comfort?: Comfort;
+  options?: TrainScheduleOptions | null;
+  power_restriction_ranges?: PowerRestrictionRange[] | null;
+  arrival_time: number;
+  mechanical_energy_consumed: {
+    base?: number;
+    eco?: number;
+  };
+  stops_count?: number;
+  path_length?: number;
```
Comment (clarani) [inline front/src/common/api/osrdEditoastApi.ts:1752]: Is it normal that all these fields are possibly undefined ?

File: front/src/reducers/osrdsimulation/index.ts [hunk 29]
```diff
@@ -155,7 +160,9 @@ export default function reducer(inputState: OsrdSimulationState | undefined, act
         );
         draft.displaySimulation =
           draft.simulation.present?.trains.length > 0 &&
-          draft.simulation.present.trains[state.selectedTrain] !== undefined;
+          // draft.simulation.present.trains[state.selectedTrain] !== undefined;
```
Comment (clarani) [inline front/src/reducers/osrdsimulation/index.ts]:
```suggestion
```

File: front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx (hunk 21 above)
Comment (SharglutDev) [reply front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx:67]: There is already an enum for the state property sent by the back end (can be seen in openapi).
I can add one in the front if you want but maybe for a fix later in another issue ?

File: front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx (hunk 22 above)
Comment (SharglutDev) [reply front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx:95]: See comment above

File: front/src/applications/stdcm/components/OldAllowances/EmptyLine.tsx (hunk 26 above)
Comment (SharglutDev) [reply front/src/applications/stdcm/components/OldAllowances/EmptyLine.tsx:42]: I get an error I remove these lines

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx (hunk 23 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]: done :)

File: front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx (hunk 24 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx]: done :)

File: front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/withOSRDData.jsx (hunk 25 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/withOSRDData.jsx]: done :)

File: front/src/applications/stdcm/views/StdcmRequestModal.jsx (hunk 27 above)
Comment (SharglutDev) [reply front/src/applications/stdcm/views/StdcmRequestModal.jsx]: done ! Nicely spotted :)

File: front/src/common/api/osrdEditoastApi.ts (hunk 28 above)
Comment (SharglutDev) [reply front/src/common/api/osrdEditoastApi.ts:1752]: I don't think so, but it should be checked with what the back returns ?

File: front/src/reducers/osrdsimulation/index.ts (hunk 29 above)
Comment (SharglutDev) [reply front/src/reducers/osrdsimulation/index.ts]: done :)

File: front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/createTrain.ts [hunk 30]
```diff
@@ -75,41 +75,37 @@ export default function createTrain(
...
+    id: train.id,
+    isStdcm: train.isStdcm,
+    name: train.name,
+    headPosition: formatStepsWithTimeMulti(train.base.head_positions),
+    tailPosition: formatStepsWithTimeMulti(train.base.tail_positions),
+    routeAspects: formatRouteAspects(train.base.route_aspects),
+    signalAspects: formatSignalAspects(train.base.signal_aspects),
+    speed: formatStepsWithTime(train.base.speeds),
+    // trainNumber is not used anymore now that we use the train id, but it is still required for TS
+    trainNumber: 0,
```
Comment (nicolaswurtz) [inline front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/createTrain.ts]: Find where it's needed and remove it please !

File: front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx (hunk 21 above)
Comment (nicolaswurtz) [reply front/src/applications/stdcm/views/OSRDCStdcmConfig.tsx:67]: You're right, enum lacks from openapi and it's where it should be, so see you in the future !

**Review by nicolaswurtz — APPROVED** [review]
